### DISCUSSION TITLE
Frigate Recording Choppy!

### DISCUSSION THREAD

**Discussion by bm55** [blakeblackshear/frigate#1802] [General]
I using 2 Reolink RLC-410 POE cameras with frigate running on a laptop with external hard drive. Both cameras can record continuous videos with maximum quality, 2560*1920, 30fps, 8192kbps on the external hard drive without any issue. The playback shows no issue but recording (clips or record) using frigate, I have to reduce the quality to about 1536kbps and 15fps. If I don't change the quality, the playback shows moving objects, like person waking across driveway freeze for about split second every 2-3 seconds. I have done some extensive troubleshooting, thinking it could be the hard drive read/write speed limitation or bandwidth issue with network but so far unable to pinpoint the cause. I have turned off the recording from the camera so only frigate is recording the videos, I have also turned off the 2nd camera, and moved frigate to newer laptop with M2 SSD, usb-c/usb3, also tried frigate 0.9 but same result. Since both cameras can record without any issue on same drives with its highest quality, I don't think it's an issue with the network bandwidth or hard drive issue. Anyone has suggestion I should try next?

Here is my ffmpeg and camera config

ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    #- /dev/dri/renderD129
    - -hwaccel_output_format
    - yuv420p
  input_args:
    - -avoid_negative_ts
    - make_zero
    - -fflags
    - nobuffer
    - -flags
    - low_delay
    - -strict
    - experimental
    - -fflags
    - +genpts+discardcorrupt
    - -rw_timeout
    - "5000000"
    - -use_wallclock_as_timestamps
    - '1' 

  garage:
    ffmpeg:
      inputs:
        - path: rtmp://192.168.0.139:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=xxxxxx
          roles:
            - detect
            - clips
            #- record
        - path: rtmp://192.168.0.139:1935/bcs/channel0_sub.bcs?channel=0&stream=2&user=admin&password=xxxxx 
          roles:
            - rtmp
      
    height: 1536 #672
    width: 2048 #896
    fps: 5
    motion:
      mask:
        - 636,1455,553,1427,549,1513,631,1509
        - 1658,83,1586,0,1889,0,1846,161,1716,185,1659,130
        - 0,750,59,799,94,829,138,786,134,711,80,647,112,519,146,458,250,421,181,338,266,364,250,258,303,279,318,235,352,254,394,231,394,199,419,161,385,107,581,57,608,0,478,0,0,0,0,480
        
    clips:
      enabled: True
      pre_capture: 10
      post_capture: 10
    rtmp:
      enabled: True   
    record:
      enabled: false  
    snapshots:
      enabled: True
      bounding_box: True
      crop: True
      timestamp: True
      #height: 200
    mqtt:
      enabled: True
    objects:
      filters:
        car:
          mask: 
            - 579,1536,1475,1493,1651,1174,1268,784,856,486,276,631
            - 558,57,932,53,1000,0,565,0
    zones:
      driveway:
        coordinates: 1794,1172,1208,632,1010,477,545,519,0,646,0,1536,2048,1536,2048,1391
        filters:
          person:
            min_area: 15000
            threshold: 0.7

      front_yard:
        coordinates: 1493,947,2048,1359,2048,695,2048,500,1798,471,1479,424,949,496
        filters:
          person:
            min_area: 8000
            threshold: 0.6

https://github.com/blakeblackshear/frigate/discussions/1802

**Comment by fbcd4f50ea5350f051450323b5abfd0b** [comment]
https://github.com/blakeblackshear/frigate/issues/1799

**Comment by bm55** [comment]
I added the -f live_flv, unfortunately it didn't fix the issue.

  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
    #- /dev/dri/renderD129
    - -hwaccel_output_format
    - yuv420p
  input_args:
    - -avoid_negative_ts
    - make_zero
    - -fflags
    - nobuffer
    - -flags
    - low_delay
    - -strict
    - experimental
    - -fflags
    - +genpts+discardcorrupt
    - -rw_timeout
    - "5000000"
    - -use_wallclock_as_timestamps
    - '1' 
    - -f 
    - live_flv

**Comment by bm55** [comment]
the http stream seems to fix this issue even with 30fps and 6144 kbps main video feed recording!
https://github.com/blakeblackshear/frigate/issues/1713#issuecomment-932976305

**Reply by Minglarn** [reply]
Have you checked the delay rtmp vs http stream? 
I've got at least 2-4 sec delay. RTMP is instant vs http. 
![image](https://user-images.githubusercontent.com/58695742/136588780-d63d53df-1b4d-4608-a3d5-0bb5d11be454.png)

And the resolution is not full.. See this:
![image](https://user-images.githubusercontent.com/58695742/136589107-42d35cbb-ffe4-4ba5-b0e8-1a75a153a6ea.png)

**Reply by bm55** [reply]
I don't see any delay between the 2 feeds. 
![image](https://user-images.githubusercontent.com/7528001/136589661-5e84db1e-e591-4c85-b1f2-6d8faee33554.png)

**Reply by Minglarn** [reply]
Well thats nice... :)
What url are you using?

**Reply by bm55** [reply]
see this post https://github.com/blakeblackshear/frigate/issues/1713#issuecomment-932976305
